Add askHow, tellHow, untellHow and unachieve illocutions.

javipalanca opened the pull request:
Some illocutions to ask for a plan, tell a plan or untell a plan have been added.
Some examples were also written to illustrate this.
PEP8 styling applied to some files.

jferrus committed:
Added askHow, tellHow and untellHow
This reverts commit b7e6698c4188a35d27b159783ab22e3480e7d1ce, reversing
changes made to fc15e4616b1e72505bb8b7f29bea251c8eaf5e52.

Revert "Merge branch 'master' of https://github.com/VRAIN-Projects/python-agentspeak"
This reverts commit 126181c2727c2cdac06c7a152e611c6c7d1da9b4, reversing
changes made to cbfcf196f17213ca76183cba3ea36c2ad7932dd5.

Merge branch 'master' of https://github.com/VRAIN-Projects/python-agentspeak

Style fixings and PEP8 corrections.

javipalanca committed:
New examples rewritten for readability.

niklasf commented:
Looks mostly good, but I don't like how annotations are parsed, but then converted back to strings and parsed again at runtime, with ad-hoc splits and slicing.

javipalanca commented:
Thanks @niklasf 
We'll work in fixing this as soon as posible.

jcarros committed:
Changes in string processing

Double string processing has been removed

jcarros committed:
Fix error plan_to_string

javipalanca committed:
Translated new example.

javipalanca commented:
Hi @niklasf 
Please, would you check if this version meets better the quality criteria of the project?
If so, it would be ready for the merge.

javipalanca commented:
this is just a kind reminder :)

niklasf commented on agentspeak/runtime.py:
```diff
@@ -408,37 +415,187 @@ def call(self, trigger, goal_type, term, calling_intention, delayed=False):
+        # Receive the agent that ask for the plan
+        for annotation in list(term.annots):
+            if isinstance(annotation, str):
+                if "askHow_sender" in annotation:
+                    sender_name = annotation.split("(")[1].split(")")[0]
```
Looks fragile.

niklasf commented on agentspeak/runtime.py:
```diff
@@ -408,37 +415,187 @@ def call(self, trigger, goal_type, term, calling_intention, delayed=False):
+                    if ("@" + str(plan_instance.annotation[0].functor)).startswith(label):
+                        plans_to_delete.append(plan_instance)
+        for plan_instance in plans_to_delete:
+            plan.remove(plan_instance)
```
`plan` as in the last plan in `plans`?

niklasf commented on agentspeak/runtime.py:
```diff
@@ -546,6 +711,45 @@ def run(self):
+    head = str(plan.head)
+    start = 0
+    
+    if "_X_" in head:
```
`_X_` can occur in the string representation without any special meaning.

niklasf commented on agentspeak/runtime.py:
```diff
@@ -546,6 +711,45 @@ def run(self):
+                head = head[:first_open+1] + str(plan.args[1]).split("(")[1].split(")")[0] + head[first_close:]
+            else:
+                head = head[:first_open+1] + str(plan.args[1]) + head[first_close:]
+            start = head.find(")", start) +1 
```
Similarly, ad-hoc parsing like this doesn't work. For example `(` could be part of a string literal, without any special meaning.

niklasf commented on agentspeak/parser.py:
```diff
@@ -900,26 +902,35 @@ def parse_event(tok, tokens, log):
     plan.loc = event.loc
 
+    # If we find a () in the event this indicate that the trigger plan have arguments, we save them in plan.args[0]
+    if "(" in str(event):
+        plan.args[0] = str(event).split("(")[1].split(")")[0]
```
Same.

niklasf reviewed (commented):
Thanks for the reminder (and your patience). I think the new version is better, but there's still quite a bit of ad-hoc parsing that breaks on more complex inputs.

jferrus committed:
Add changes comment 1

jcarros committed:
Add changes comments

javipalanca commented:
Hi @niklasf 
Check if these commits match your quality requirements. If you need more changes don't hesitate to ask!
Thanks.

XimoTaverner committed:
Updated Agent's call funciton

Modified the call function to fix plan application when a belief is added or removed

jferrus committed:
Merge pull request #1 from XimoTaverner/master

Updated Agent's call funciton
